build/sslstream-scenarios.yml

```diff
@@ -0,0 +1,88 @@
+# SslStream scenarios
+
+parameters:
+  - name: arguments
+    type: string
+    default: ""
+  - name: connection
+    type: string
+    default: ""
+  - name: condition
+    type: string
+    default: "true"
+
+  - name: readWriteScenarios
+    type: object
+    default:
+      - displayName: "read-write"
+        arguments: --scenario read-write --client.options.collectCounters true
+
+  - name: handshakeScenarios
+    type: object
+    default:
+      - displayName: "read-write"
+        arguments: --scenario handshake --client.options.collectCounters true
+
+  - name: tlsResume
+    type: object
+    default:
+      - displayName: "TLS Resume"
+        arguments: --variable allowTlsResume=true --property tlsResume=resume
+      - displayName: "No TLS Resume"
+        arguments: --variable allowTlsResume=false --property tlsResume=no-resume
+
+  - name: tlsVersion
+    type: object
+    default:
+      - displayName: "TLS 1.3"
+        arguments: --variable tlsVersion=1.3 --property tlsVersion=1.3
+      - displayName: "TLS 1.2"
+        arguments: --variable tlsVersion=1.2 --property tlsVersion=1.2
+
+  - name: concurrency
+    type: object
+    default:
+      - displayName: "1 connection"
+        arguments: --variable concurrency=1 --property concurrency=1
+      - displayName: "16 connections"
+        arguments: --variable concurrency=16 --property concurrency=16
+
+steps:
+  # handshake scenarios
+  - ${{ each s in parameters.handshakeScenarios }}:
+      - ${{ each version in parameters.tlsVersion }}:
+          - ${{ each res in parameters.tlsResume }}:
+              - ${{ each c in parameters.concurrency }}:
+                  - task: PublishToAzureServiceBus@1
+                    condition: succeededOrFailed()
+                    timeoutInMinutes: 10
+                    displayName: ${{ version.displayName }} ${{ s.displayName }} ${{ res.displayName }}
+                    inputs:
+                      connectedServiceName: ${{ parameters.connection }}
+                      waitForCompletion: true
+                      useDataContractSerializer: "false"
+                      messageBody: |
+                        {
+                          "condition": "(${{ parameters.condition }}) && ((new Date().getUTCHours() - 7 + 24) % 24 >= 13)",
+                          "name": "crank",
+                          "args": [ "--client.framework net9.0 --server.framework net9.0 --command-line-property --table HttpClientBenchmarks --sql SQL_CONNECTION_STRING --chart --session $(session) ${{ parameters.arguments }} --no-metadata --no-measurements ${{ s.arguments }} ${{ version.arguments }} ${{ res.arguments }} ${{ c.arguments }}" ]

```

Comment:
The existing benchmarks are already stored in the HttpClientBenchmarks table but these ones have different properties so this would make sense to use a distinct table, and even more if they have to show up in a different PBI page with custom filters (concurrency, version, ...).

e.g., SslStreamBenchmarks

Reply:
Please add this argument `--description \"${{ version.displayName }} ${{ s.displayName }} ${{ res.displayName }} ${{ c.displayName }} $(System.JobDisplayName)\" --property buildId=\"$(buildId)\" --property buildNumber=\"$(buildNumber)\"` which will allow us to add regression detection.

I use this field and the scenario name to group all runs (by unique combination of arguments).

Reply:
also, it seems that `$(sslStreamJobs)` is not added anywhere, we'd need to pass it as well so the `--config` argument would be specified in the resulting command line

Reply:
@CarnaViire it's coming from job-variables.yml which is added in the main ADO template.